api.go

```diff
@@ -32,9 +31,15 @@ const (
 	API_ERR_INTERNAL
 )
 
+type deploymentsResult struct {
+	deployments []DataDeployment
+	err         error
+}
+
 var (
-	deploymentsChanged = make(chan string)
-	addSubscriber      = make(chan chan string)
+	deploymentsChanged = make(chan interface{})
```

Comment:
@theganyo the `send` routine is a blocking routine, cause it's trying to send over `deliverDeployments` channel.
If `send` is blocked, the debounce routine will be blocked. `debounce` is also a receiver of the `deploymentsChanged` channel, so it will also block the senders of this channel.
So i would suggest use some buffered channels to alleviate the potential blocking chains and increase concurrency.